pkg/resource/predicates.go

```diff
@@ -76,6 +76,34 @@ func HasProvisioner(c client.Client, cr ClassReferencer, provisioner string) boo
 	return strings.EqualFold(cs.Provisioner, provisioner)
 }
 
+// NOTE(hasheddan): HasClassReferenceKind should eventually replace ObjectHasProvisioner
+// when strongly typed resource classes are implemented
+
+// HasClassReferenceKind accepts ResourceClaims that reference the correct kind of resourceclass
+func HasClassReferenceKind(k ClassKind) PredicateFn {
+	return func(obj runtime.Object) bool {
+		cr, ok := obj.(ClassReferencer)
+		if !ok {
+			return false
+		}
+
+		ref := cr.GetClassReference()
+		if ref == nil {
+			return false
+		}
+
+		gvk := ref.GroupVersionKind()
+
+		if gvk.Group != k.Group {
```

Comment:
is there not an equality implementation already implemented upstream for `GroupVersionKind`?

Reply:
Because `GroupVersionKind` is a struct with three fields, each of type `string`, we should be able to compare using the GVK's with `==` per the spec [here](https://golang.org/ref/spec#Comparison_operators).
> Struct values are comparable if all their fields are comparable. Two struct values are equal if their corresponding non-blank fields are equal.

Does that sound correct to you?

Reply:
Updated in most recent commit :)